lib/logstash/inputs/decoder_impl.rb

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
```

Comment:
We tend to put plugin-specific code in this pattern of file name:

```/lib/logstash/TYPE/NAME/xyz.rb```

For this, I mean: `lib/logstash/inputs/tcp/decoder_impl.rb`